src/zap/zapinfo.cpp

```diff
@@ -1736,6 +1736,15 @@ void* ZapInfo::getTailCallCopyArgsThunk (
     return m_pImage->GetWrappers()->GetStub(pStub);
 }
 
+bool ZapInfo::getTailCallHelp(
+        CORINFO_METHOD_HANDLE hTarget,
+        CORINFO_SIG_INFO* callSiteSig,
+        CORINFO_GET_TAILCALL_HELP_FLAGS flags,
+        CORINFO_TAILCALL_HELP* pResult)
+{
+    return false;
```

Comment:
I don't know if this is ok either.

Reply:
This should rather throw exception, e.g. `ThrowHR(E_NOTIMPL)`. There are number of other similar places.

The behavior we want is: If the IL contains explicit tail. prefix, we are precompiling the code and not able to do fast tail call, we should leave a hole in the image and fallback to JIT at runtime that is able to generate the slow tail call.

Reply:
Should the crossgen2 implementation do the same?

Reply:
Yes.